packages/components/src/composite/index.ts

```diff
@@ -1,7 +1,135 @@
-// Originally this pointed at a Reakit implementation of
-// `Composite`, but we are removing Reakit entirely from the
-// codebase. We will continue to support the Reakit API
-// through the 'legacy' version, which uses Ariakit under
-// the hood.
+/**
+ * Composite is a component that may contain navigable items represented by
+ * CompositeItem. It's inspired by the WAI-ARIA Composite Role and implements
+ * all the keyboard navigation mechanisms to ensure that there's only one
+ * tab stop for the whole Composite element. This means that it can behave as
+ * a roving tabindex or aria-activedescendant container.
+ *
+ * @see https://ariakit.org/components/composite
+ */
 
-export * from './legacy';
+/**
+ * External dependencies
+ */
+import {
+	Composite,
+	CompositeGroup,
+	CompositeGroupLabel,
+	CompositeItem,
+	CompositeRow,
+	useCompositeStore,
+} from '@ariakit/react';
+import type { CompositeStore, CompositeStoreProps } from '@ariakit/react';
+
+/**
+ * Creates a composite store.
+ * @see https://ariakit.org/components/composite
+ * @example
+ * ```jsx
+ * const useCompositeStore = Composite.useStore;
+ * const store = useCompositeStore();
+ * <Composite.Root store={store}>
+ *   <Composite.Item>Item</Composite.Item>
+ *   <Composite.Item>Item</Composite.Item>
+ *   <Composite.Item>Item</Composite.Item>
+ * </Composite.Root>
+ * ```
+ */
+export const useStore = useCompositeStore;
+
+/**
+ * Renders a composite widget.
+ * @see https://ariakit.org/components/composite
+ * @example
+ * ```jsx
+ * const useCompositeStore = Composite.useStore;
+ * const store = useCompositeStore();
+ * <Composite.Root store={store}>
+ *   <Composite.Item>Item 1</Composite.Item>
+ *   <Composite.Item>Item 2</Composite.Item>
+ * </Composite.Root>
+ * ```
+ */
+export const Root = Composite;
```

Comment:
I see the need to have a `Composite.Root`, but I kind of can't really love that name. I wonder what other alternatives we have. Sometimes I wish we could just use `Composite`.

Reply:
I understand your point! That would also be my preference if we could start a new package from scratch.

But as explained before, our decision was influenced by the fact that we have many older, monolithic components in the package. Associating dot notation to compound components brings to important benefits:

- a clear distinction in compound vs monolithic components for consumers of the package;
- a way to avoid any potential naming conflicts, in case a component exposes both a compound and a monolithic version